packages/tempus/src/Tempus_RKButcherTableau.hpp

```diff
@@ -285,9 +188,105 @@ class RKButcherTableau :
       if (nonZero == false) isDIRK_ = false;
     }
 
-    Teuchos::RCP<Teuchos::ParameterList>   RK_stepperPL_;
+    virtual void setAbc(
+      const Teuchos::SerialDenseMatrix<int,Scalar>& A,
+      const Teuchos::SerialDenseVector<int,Scalar>& b,
+      const Teuchos::SerialDenseVector<int,Scalar>& c,
+      const int order,
+      const std::string& longDescription,
+      const Teuchos::SerialDenseVector<int,Scalar>&
+        bstar = Teuchos::SerialDenseVector<int,Scalar>())
+    { setAbc(A,b,c,order,order,order,longDescription,bstar); }
 
-  private:
+    virtual void setAbc(
+      const Teuchos::SerialDenseMatrix<int,Scalar>& A,
+      const Teuchos::SerialDenseVector<int,Scalar>& b,
+      const Teuchos::SerialDenseVector<int,Scalar>& c,
+      const int order,
+      const int orderMin,
+      const int orderMax,
+      const std::string& longDescription,
+      const Teuchos::SerialDenseVector<int,Scalar>&
+        bstar = Teuchos::SerialDenseVector<int,Scalar>())
+    {
+      const int numStages = A.numRows();
+      TEUCHOS_ASSERT_EQUALITY( A.numCols(), numStages );
+      TEUCHOS_ASSERT_EQUALITY( b.length(), numStages );
+      TEUCHOS_ASSERT_EQUALITY( c.length(), numStages );
+      TEUCHOS_ASSERT( order > 0 );
+      A_ = A;
+      b_ = b;
+      c_ = c;
+      order_ = order;
+      orderMin_ = orderMin;
+      orderMax_ = orderMax;
+      this->set_isImplicit();
+      this->set_isDIRK();
+
+      // Consistency check on b
+      Scalar sumb = Scalar(0.0);
+      for (size_t i = 0; i < this->numStages(); i++) sumb += b_(i);
+      TEUCHOS_TEST_FOR_EXCEPTION( std::abs(Scalar(1.0)-sumb) > 1.0e-08,
```

Comment:
Would you consider picking this tolerance as a function of Scalar, say by using `Teuchos::ScalarTraits`?

Reply:
Hi Mark,

Not sure if I understand you.  I probably should have used Teuchos::ScalarTraits for Scalar(1.0).  Are you suggesting using Teuchos::ScalarTraits for the 1.0e-08?

Reply:
@ccober6 You could use `Teuchos::ScalarTraits<Scalar>::epsilon()` (or perhaps the square root of that).  `Scalar(1.0)` is fine.  (`Scalar(1)` might not be, for complex `Scalar`.)

Reply:
It doesn't really matter if you don't test for `float`, though.